### PR TITLE
[Networking] Combined all netlink-related processing into SocketNetlink

### PR DESCRIPTION
**Pull request by mipo57**
- Moved every Netlink-related processing into a NetlinkSocket class.
- Added possibility to call exchange asynchronously
- Multi message packets are now parsed correctly
- Updated frame parsing to use NLMSG macros to make sure padding is correct

### CODE REVIEW

**Comment by mipo57** [comment]
Still more work on making networking asynchronous needs to be done. Possibility to use Netlink asynchronously was added to SocketNetlink, however, NetworkControl plugin is still written in a synchronous way. At the moment, if something goes wrong  (especially during event-based flushing) we risk blocking ResourceMonitor thread.

File: Source/core/Netlink.h [hunk 1]
```diff
@@ -396,7 +437,7 @@ namespace Core {
 
     public:
         SocketNetlink(const Core::NodeId& destination)
-            : SocketDatagram(false, destination, Core::NodeId(), 512, 1024)
+            : SocketDatagram(false, destination, Core::NodeId(), 4096, 8192)
```
Comment (pwielders) [inline Source/core/Netlink.h:443]: Did you get the sizes from a spec? Sending 4K and Receiving 8K? sounds big... Maybe good to put the reference here where you found this..

File: Source/core/NetworkInfo.cpp [hunk 2]
```diff
@@ -1258,35 +1232,109 @@ namespace Core {
...
+
+        return (_channel->Exchange(modifier, modifier, waitTime));
     }
 
     uint32_t IPNetworks::Network::Gateway(const IPNode& network, const NodeId& gateway)
     {
         IPRouteModifyType<true> modifier(*this, network, gateway);
 
-        return (_channel->Exchange(modifier, modifier));
+#ifdef __DEBUG__
```
Comment (pwielders) [inline Source/core/NetworkInfo.cpp]: Move this waitTime definition used (and now copied multiple times :-) ) to a single class static constexpr...

**Review by pwielders — CHANGES_REQUESTED** [review]
Looks good just some minor remarks (create 1 setting for waitTime, that is different in debug time), and a general question. I assume you did not change any interfaces that are publicly used (I saw one change from Size() -> PayloadSize()) but I am hoping that is all.
The reason why I am asking:
1) Is windows still working :-)
2) The Netlink socket is used in projects you are not building and/or testing, should I rebuild/retest and fix those?

File: Source/core/Netlink.h (hunk 1 above)
Comment (mipo57) [reply Source/core/Netlink.h:443]: I don't think they are limited by spec, however, in practice, it is limited to page size. Moreover, recommendation is to limit it to 8k if page size > 8k (https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/include/linux/netlink.h?h=linux-4.9.y#n112). I limited send buffer to 4k as we control it and can always serve a more restricted 4k. I will add it in the comment :)

Comment (mipo57) [reply Source/core/Netlink.h:443]: In practice, I found multiple cases where the message was >4k and split into 2 ~3.5k  (as RPI page size is 4k). I think limiting it to 4k would be a little bit on the dangerous side here

File: Source/core/NetworkInfo.cpp (hunk 2 above)
Comment (mipo57) [reply Source/core/NetworkInfo.cpp]: 2 Changes were made. The Size() -> PayloadSize() you mentioned and Data() -> template<> Payload(), however, both changes were made in Frames class that is used to parse Netlink messages and wasn't used anywhere else (event in code you provided in conversation). 

These changes are obviously not absolutely required, however it was confusing at times whether Data() and Size() were referring to just payload or the whole frame.

Comment (mipo57) [reply Source/core/NetworkInfo.cpp]: Actually after reviewing one more thing changed in Network (Event class), however, it came mostly from cutting changes into this smaller PR and I will undo it

**Review by pwielders — APPROVED** [review]

**Pull request merged**